src/evaluators/ForInStatement.js

```diff
@@ -133,6 +133,7 @@ function emitResidualLoopIfSafe(
     let absStr = realm.createAbstract(
       new TypesDomain(StringValue),
       ValuesDomain.topVal,
+      0,
```

Comment:
How about a special hash value or a flag, and assert that this never hits the serializer?